planner/core/optimizer.go

```diff
@@ -66,6 +66,7 @@ type logicalOptRule interface {
 func BuildLogicalPlan(ctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (Plan, error) {
 	ctx.GetSessionVars().PlanID = 0
 	ctx.GetSessionVars().PlanColumnID = 0
+	ctx.GetSessionVars().StmtCtx.UseCache = false
```

Comment:
`StmtCtx.UseCache` is set in: https://github.com/pingcap/tidb/blob/master/executor/prepared.go#L155

Then the function `BuildLogicalPlan()` is called. Maybe it's better to set `StmtCtx.UseCache` to false before the function `BuildLogicalPlan()` call: https://github.com/pingcap/tidb/blob/master/executor/prepared.go#L162

Reply:
If we can ensure calling `BuildLogicalPlan()`  at `prepared.go#L162` only, setting the variable at the location as you said is good. Otherwise, setting the variable in `BuildLogicalPlan()` might be a good choice in order to prevent from a potential risk because `StmtCtx.UseCache` should be false in the call paths of `BuildLogicalPlan()`.

Reply:
Yes, we can ensure. Function `BuildLogicalPlan()` is only used in the test code and the prepare statements:
<img width="1164" alt="2018-10-19 13 03 17" src="https://user-images.githubusercontent.com/5268763/47198543-660a1e00-d39f-11e8-9d93-4380b8d33a49.png">

Reply:
You can get the similar result by:
```sh
find . -name "*.go" | xargs grep --color -n "\<BuildLogicalPlan\>"
```